modules/concom/sitesupport/components/staff-department.js

```diff
@@ -0,0 +1,88 @@
+/* globals Vue, apiRequest */
+import { extractDepartmentStaff } from '../department-staff-parser.js';
+
+const PROPS = {
+  department: Object,
+  division: Object,
+  currentUser: Object
+}
+
+const TEMPLATE = `
+  <div class="UI-table UI-table-heading" v-if="isDepartment(department).value">
+    <div class="UI-table-row event-color-primary">
+      <div class="UI-table-cell-no-border">{{department.name}}</div>
+      <div class="UI-table-cell-no-border">
+        <a :href="divisionNavigationRef(division).value">{{division.name}}</a>
+      </div>
+      <div class="UI-table-cell-no-border">
+        <template v-for="email in department.email">{{email}}<br/></template>
+      </div>
+    </div>
+  </div>
+  <div :class="tableClass(department).value">
+    <department-header :isDepartment=isDepartment(department).value :department=department></department-header>
+    <div class="UI-table-row" v-for="staff in departmentStaff">
+      <department-member :staff=staff :isDepartment=isDepartment(department).value :currentUser=currentUser></department-member>
+    </div>
+  </div>
+`;
+
+const DIVISION_POSITIONS = ['Head', 'Specialist'];
```

Comment:
I think that's a good idea in the long run. I would rather be able to get the possible positions from an API call and use them in the templates instead of having hard-coded values for something that may potentially change.

The tricky part I think is that right now, we have no indicator on that table whether a position is supposed to show up for a division or not - at the moment I think it was coded separately specifically for `Head` and `Specialist` positions both in the original and in the component version.

How would you feel about this as a short-term compromise? For the sake of keeping PRs smaller, can we let it slide in the PR, and I can quickly follow up with a new PR that includes an API built to expose the `ConComPositions` entries? I can also switch over the implementation in the new PR.

I think in the API PR, I would probably also like to modify the table in some way so that we can easily use the values in the UI. There are a couple ways to achieve this probably, and maybe the simpler one is to add a new column that includes the name that should be used for divisions on each position. It would be a nullable field since a value of `null` or `''` indicates that it should be omitted from division display entries.

What are your thoughts on this?

Reply:
I can buy that.